### PR TITLE
Refactor to add `selectorSourceIndex` utility

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

See : https://github.com/stylelint/stylelint/pull/7685#discussion_r1597462061

> Is there anything in the PR that needs further explanation?

I had actually expected this to be simpler but I now understand that this isn't possible.

The whitespace between `,` and the selector needs to go somewhere in the AST and the selector node is the right place. This implies that a selector `sourceIndex` includes leading whitespace.

We want to find the `sourceIndex` value without leading whitespace.

`selectorNode.first?.sourceIndex ?? 0`: this was still not ideal because `0` is not the right fallback.

For example with : `.foo,,`

This is a list with 3 entries. The middle entry is definitely empty.
We don't want to use `0` as the source index value because that is the location of `.` in `.foo`.

Using `node.first?.sourceIndex ?? node.sourceIndex` is more correct.

This also updates `postcss-selector-parser` to `6.1.0` as this version includes the patches to make `node.sourceIndex` available.

### CODE REVIEW

File: lib/utils/selectorSourceIndex.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Get the source index of a selector, excluding leading whitespace.
+ *
+ * @typedef {import('postcss-selector-parser').Container<string | undefined>} Container
+ * @param {Container} node
+ *
+ * @returns {number}
+ */
+export default function selectorSourceIndex(node) {
```
Comment (Mouvedia) [inline lib/utils/selectorSourceIndex.mjs:9]: question
nit

Are all type of `node`s accepted?

**Review by Mouvedia — APPROVED** [review]

File: lib/utils/selectorSourceIndex.mjs
Comment (ybiquitous) [inline lib/utils/selectorSourceIndex.mjs:1]: [suggest] For consistency, how about renaming the function to `getSelectorSourceIndex` like other `get*` utils (e.g. `getAtRuleParams`)? But this suggestion depends on my feelings, and you can leave the name as-is if you have any thoughts. 😅

**Review by ybiquitous — APPROVED** [review]
Nice catch! I left a comment on styling, but LGTM 👍🏼

File: lib/utils/selectorSourceIndex.mjs
Comment (romainmenke) [reply lib/utils/selectorSourceIndex.mjs:1]: Thank you!

Naming these things is always tricky for me, so this feedback is very much appreciated!

**Review by ybiquitous — APPROVED** [review]
LGTM!

**Comment by romainmenke** [comment]
Thank you for the reviews 🙇

**Pull request merged**